maven-javadoc-plugin 3.2.0 breaks release

I was repeatedly unable to successfully execute `mvn release:perform` with the latest version of the javadoc plugin.  Rolling back to 3.1.1 caused no problems.  I can't see anything obvious about what might have changed and would appreciate the help of someone more experienced in maven trying to troubleshoot this.

It works fine on the `oshi-core` artifact but fails on the shaded version.

Log right before the errors posted below.

```
    Building index for all the packages and classes...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/overview-tree.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/deprecated-list.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/index.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/index-all.html...
    Building index for all classes...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/allclasses-index.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/allpackages-index.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/overview-summary.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/help-doc.html...
    1 error
    [INFO] ------------------------------------------------------------------------
    [INFO] Reactor Summary for Operating System and Hardware Information 4.5.1:
    [INFO] 
    [INFO] Operating System and Hardware Information .......... SUCCESS [01:16 min]
    [INFO] oshi-core .......................................... SUCCESS [04:42 min]
    [INFO] oshi-core-shaded ................................... FAILURE [  6.868 s]
    [INFO] oshi-demo .......................................... SKIPPED
    [INFO] oshi-dist .......................................... SKIPPED
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time:  06:05 min
    [INFO] Finished at: 2020-03-19T16:40:04-07:00
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (javadoc-jar) on project oshi-core-shaded: MavenReportException: Error while generating Javadoc: 
    [ERROR] Exit code: 1 - javadoc: error - Error reading file: /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/../apidocs/element-list
    [ERROR] 
    [ERROR] Command line was: /usr/local/opt/openjdk/bin/javadoc @options @packages
```

Well, I didn't even get as far as `mvn release:perform`.  It failed on the `release:prepare` step:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (attach-javadocs) on project oshi-demo: MavenReportException: Error while generating Javadoc: 
    [ERROR] Exit code: 1 - error: module org.slf4j.simple reads package oshi from both com.github.oshi.demo and com.github.oshi
    [ERROR] error: module org.slf4j.simple reads package oshi.jna.platform.windows from both com.github.oshi.demo and com.github.oshi
    [ERROR] error: module org.slf4j.simple reads package oshi.jna.platform.linux from both com.github.oshi.demo and com.github.oshi
    [ERROR] error: module org.slf4j.simple reads package oshi.jna.platform.unix from both com.github.oshi.demo and com.github.oshi
<and lots more like that>
```
But I haven't declared any modules, just added the "Automatic Module Name".   

If there's an easy fix to this, I'd welcome it, otherwise I'm leaning toward completely removing anything related to Java 9+ until someone who actually needs it submits a PR.

Added `<source>8</source>`.  New error. Building with jdk13.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:javadoc (default-cli) on project oshi-core: An error has occurred in Javadoc report generation: 
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses packages in the unnamed module, but the packages defined in https://docs.oracle.com/en/java/javase/13/docs/api/ are in named modules.
```

And, nope.  Still fails on 3.2.0 even with a source path pointing to oshi-core.
```
    [INFO] No previous run data found, generating javadoc.
    [INFO] 
    Loading source file /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/../oshi-core/src/main/java/oshi/hardware/CentralProcessor.java...
```
... loaded all the source files from `oshi-core` source directory
```
    Constructing Javadoc information...
    Standard Doclet version 15.0.1+9
    Building tree for all the packages and classes...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/oshi/driver/unix/aix/Ls.html...
```
... Generated html in `oshi-core-shaded/target/apidocs`
```
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/overview-tree.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/deprecated-list.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/index.html...
    Building index for all the packages and classes...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/index-all.html...
    Building index for all classes...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/allclasses-index.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/allpackages-index.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/overview-summary.html...
    Generating /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/help-doc.html...
    1 error
```
And chokes:
```
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (javadoc-jar) on project oshi-core-shaded: MavenReportException: Error while generating Javadoc: 
    [ERROR] Exit code: 1 - javadoc: error - Error reading file: /Users/danielwiddis/git/oshi/target/checkout/oshi-core-shaded/target/apidocs/../apidocs/element-list
```
So I guess it's back to 3.1.1.

Well, JNA 5.9.0 came out so I attempted a release today and hit a new problem:
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.9.1:site (default-site) on project oshi-core: Error generating maven-javadoc-plugin:3.1.1:aggregate-no-fork report: 
> [ERROR] Exit code: 1 - /Users/danielwiddis/git/oshi/oshi-core/src/main/java/oshi/hardware/platform/mac/MacComputerSystem.java:31: error: package com.sun.jna does not exist
> [ERROR] import com.sun.jna.Native; 

This occurs during the `mvn site` step.  It looks like it may be a manifestation of https://issues.apache.org/jira/browse/MJAVADOC-609 which also points to https://issues.apache.org/jira/browse/MJAVADOC-620 as a root cause, but these were allegedly fixed in 3.2.0.

But I tried with 3.3.0 and the same issue.  I tried rolling back to the previous release state and the same issue.

Only thing that I think changed is my jdk version (building with 16.0.2).   Guess I'll try an earlier version there....

--- a/oshi-core-shaded/pom.xml
+++ b/oshi-core-shaded/pom.xml
@@ -22,7 +22,6 @@
 
     <properties>
         <main.basedir>${project.parent.basedir}</main.basedir>
-        <automatic.module.name>com.github.oshi.shaded</automatic.module.name>
     </properties>
 
     <dependencies>
@@ -76,7 +75,6 @@
                             <goal>jar</goal>
                         </goals>
                         <configuration>
-                            <sourcepath>../oshi-core/src/main/java</sourcepath>
                             <includeDependencySources>true</includeDependencySources>
                             <dependencySourceIncludes>
                                 <!-- include ONLY dependencies I control -->

--- a/oshi-core/pom.xml
+++ b/oshi-core/pom.xml
@@ -21,17 +21,14 @@
     </scm>
 
     <properties>
-        <!-- Users of the Spring Boot Starter Parent should include this property
-            in their POM -->
+        <!-- Users of the Spring Boot Starter Parent should include this property in their POM -->
         <jna.version>5.10.0</jna.version>
         <main.basedir>${project.parent.basedir}</main.basedir>
-        <automatic.module.name>com.github.oshi</automatic.module.name>
     </properties>
 
     <dependencyManagement>
         <dependencies>
-            <!-- Due to critical nature of OSHI jna usage, set to dependency management
-                to help influence usage -->
+            <!-- Due to critical nature of OSHI jna usage, set to dependency management to help influence usage -->
             <dependency>
                 <groupId>net.java.dev.jna</groupId>
                 <artifactId>jna</artifactId>
@@ -77,4 +74,20 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <Automatic-Module-Name>com.github.oshi</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 </project>

--- a/oshi-demo/pom.xml
+++ b/oshi-demo/pom.xml
@@ -24,7 +24,6 @@
         <main.basedir>${project.parent.basedir}</main.basedir>
         <jackson.version>2.13.1</jackson.version>
         <jfreechart.version>1.5.3</jfreechart.version>
-        <automatic.module.name>com.github.oshi.demo</automatic.module.name>
     </properties>
 
     <dependencies>

--- a/oshi-dist/pom.xml
+++ b/oshi-dist/pom.xml
@@ -20,7 +20,6 @@
 
     <properties>
         <main.basedir>${project.parent.basedir}</main.basedir>
-        <automatic.module.name>com.github.oshi.dist</automatic.module.name>
     </properties>
 
     <!-- NOTE: These dependency declarations are only required to sort this project 

--- a/pom.xml
+++ b/pom.xml
@@ -125,7 +125,7 @@
         <maven-checkstyle-plugin.version>3.1.2</maven-checkstyle-plugin.version>
         <checkstyle.config.location>config/checkstyle.xml</checkstyle.config.location>
         <checkstyle.suppressions.location>config/checkstyle-suppressions.xml</checkstyle.suppressions.location>
-        <maven-javadoc-plugin.version>3.1.1</maven-javadoc-plugin.version>
+        <maven-javadoc-plugin.version>3.3.1</maven-javadoc-plugin.version>
         <!-- tools -->
         <maven-antrun-plugin.version>3.0.0</maven-antrun-plugin.version>
         <maven-assembly-plugin.version>3.3.0</maven-assembly-plugin.version>
@@ -294,7 +294,7 @@
                                 <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
                                 <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
                             </manifest>
-                            <manifestEntries>
+                            <manifestEntries combine.children="append">
                                 <Build-Time>${maven.build.timestamp}</Build-Time>
                                 <Copyright>${copyright}</Copyright>
                                 <Git-Revision>${git.commit.id}</Git-Revision>
@@ -303,7 +303,6 @@
                                 <Os-Version>${os.version}</Os-Version>
                                 <X-Compile-Source-JDK>${maven.compiler.source}</X-Compile-Source-JDK>
                                 <X-Compile-Target-JDK>${maven.compiler.target}</X-Compile-Target-JDK>
-                                <Automatic-Module-Name>${automatic.module.name}</Automatic-Module-Name>
                             </manifestEntries>
                         </archive>
                     </configuration>
@@ -417,14 +416,11 @@
                         </archive>
                     </configuration>
                     <executions>
-                        <!-- ! here we override the super-pom attach-sources execution 
-                            id which ! calls sources:jar goal. That goals forks the lifecycle, ! causing the 
-                            generate-sources phase to be called twice for the install goal. ! Starting with 
-                            Maven 3.4.0 (https://issues.apache.org/jira/browse/MNG-5940) ! this is not needed 
-                            anymore. -->
-                        <!-- except that OSSRH fails with no sources with this 
-                            excluded <execution> <id>attach-sources</id> <phase>DISABLE_FORKED_LIFECYCLE_MSOURCES-13</phase> 
-                            </execution> -->
+                        <!-- Here we override the super-pom attach-sources execution id which calls sources:jar goal. That 
+                            goals forks the lifecycle, causing the generate-sources phase to be called twice for the install goal. Starting with Maven 
+                            3.4.0 (https://issues.apache.org/jira/browse/MNG-5940) this is not needed anymore. -->
+                        <!-- except that OSSRH fails with no sources with this excluded <execution> <id>attach-sources</id> 
+                            <phase>DISABLE_FORKED_LIFECYCLE_MSOURCES-13</phase> </execution> -->
                     </executions>
                 </plugin>
                 <!-- External Tools -->
@@ -636,8 +632,7 @@
                                         module path</reason>
                                     <includeTestCode>true</includeTestCode>
                                     <bannedImports>
-                                        <!-- Disallow all imports except explicitly 
-                                            allowed -->
+                                        <!-- Disallow all imports except explicitly allowed -->
                                         <bannedImport>**</bannedImport>
                                     </bannedImports>
                                     <allowedImports>
@@ -924,8 +919,8 @@
             <build>
                 <pluginManagement>
                     <plugins>
-                        <!--This plugin's configuration is used to store Eclipse 
-                            m2e settings only. It has no influence on the Maven build itself. -->
+                        <!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on 
+                            the Maven build itself. -->
                         <plugin>
                             <groupId>org.eclipse.m2e</groupId>
                             <artifactId>lifecycle-mapping</artifactId>
@@ -1046,8 +1041,8 @@
                                 <configuration>
                                     <keyname>${gpg.keyname}</keyname>
                                     <passphraseServerId>${gpg.keyname}</passphraseServerId>
-                                    <!-- GPG 2.1 requires pinentry-mode to be set 
-                                        to loopback in order to pick up the gpg.passphrase value defined in Maven settings.xml. -->
+                                    <!-- GPG 2.1 requires pinentry-mode to be set to loopback in order to pick up the gpg.passphrase 
+                                        value defined in Maven settings.xml. -->
                                     <gpgArguments>
                                         <arg>--pinentry-mode</arg>
                                         <arg>loopback</arg>

